dascore/io/segy/utils.py

```diff
@@ -3,14 +3,80 @@
 from __future__ import annotations
 
 import datetime
+import warnings
 
 import numpy as np
-from segyio import TraceField
+
+# --- Getting format/version
+import pandas as pd
 
 import dascore as dc
+from dascore import to_float
 from dascore.core import get_coord_manager
+from dascore.exceptions import InvalidSpoolError, PatchError
+from dascore.utils.misc import optional_import
+
+# Valid data format codes as specified in the SEGY rev1 manual.
+VALID_FORMATS = [1, 2, 3, 4, 5, 8]
+
+# This is the maximum possible interval between two samples due to the nature
+# of the SEG Y format.
+MAX_INTERVAL_IN_SECONDS = 0.065535
+
+# largest number possible with int16
+MAX_NUMBER_OF_SAMPLES = 32767
```

Comment:
Do we need these?

Reply:
It looks like we don't since they are edge cases. But for the writer, we may want to ensure we don't make a trace with more than max number of samples. Since DAS can be very small dt and very large time, this could become an issue.